### PR TITLE
[+] Command Generate - Add logs to explain what is going on

### PR DESCRIPTION
**Pull request by VincentMolinie**
It gives something like that
Capture d’écran 2019-08-06 à 13 42 59

### CODE REVIEW

**Comment by VincentMolinie** [comment]
[P3 - Snake Case DB - When using snaked_case in db forest-express-sequelize doesn't handle well columnName](https://trello.com/c/PPGCVuIG/3571-p3-snake-case-db-when-using-snakedcase-in-db-forest-express-sequelize-doesnt-handle-well-columnname)

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,9 +1,17 @@
...
+- Command Generate - Add logs to explain what is going on.
+
+### Changed
+- Command Generate - Camelize all field names.
+- Command Generate - Camelize all reference names and remove `id` or `uuid` from name.
+
 ### Fixed
 - Command Generate - Fix package name.
 - Command Generate - Remove trailing spaces in `.env`.
+- Command Generate - Do not duplicate `createAt` and `updatedAt` if not named as camelCase in database.
```
Comment (arnaudbesnier) [inline CHANGELOG.md]: `createAt` => `createdAt`.
Do you take the time to review your own PR?
Such typo should be detected by ourself, not your reviewer.

File: services/database-analyzer.js [hunk 2]
```diff
@@ -37,28 +48,49 @@ function DatabaseAnalyzer(databaseConnection, config) {
...
             if (!(columnName === 'id' && type === 'INTEGER' && columnInfo.primaryKey)) {
               const field = {
+                cameCaseName: _.camelCase(columnName),
                 name: columnName,
                 type,
                 primaryKey: columnInfo.primaryKey,
                 defaultValue: columnInfo.defaultValue,
               };
 
+              const typeOfField = field.primaryKey ? 'primary key' : 'field';
```
Comment (arnaudbesnier) [inline services/database-analyzer.js]: Isn't a primary key also a field?
if you want to add the primary key information, it should be additional.

File: services/database-analyzer.js [hunk 3]
```diff
@@ -127,10 +159,9 @@ function DatabaseAnalyzer(databaseConnection, config) {
             if (err.message && err.message.startsWith('CMD_NOT_ALLOWED')) {
               logger.warn(`⚠️  [${collection.name}] CMD_NOT_ALLOWED: mapReduce. Please, write manually the Mongoose fields on this collection.  ⚠️`);
               logger.warn('If your database is hosted on MongoDB Atlas, it\'s probably due to the Free tier limitations. More info here: https://docs.atlas.mongodb.com/unsupported-commands\n');
-              results = [];
-            } else {
-              return reject(err);
+              return resolve([]);
             }
+            return reject(err);
```
Comment (arnaudbesnier) [inline services/database-analyzer.js]: Good code improvement 👍🏾 => `devel`

File: templates/model.txt [hunk 4]
```diff
@@ -12,7 +12,7 @@ module.exports = mongoose.model('<%= table %>', schema, '<%= table %>');
 %>module.exports = (sequelize, DataTypes) => {
   const { Sequelize } = sequelize;
   const Model = sequelize.define('<%= table %>', {<% _.each(fields, (field) => { %>
-    '<%= field.name %>': {
+    <%= field.cameCaseName %>: {
```
Comment (arnaudbesnier) [inline templates/model.txt]: I am pretty sure, this change breaks the behaviour for schema without underscores but having field names like this `recordID`:
Screenshot 2019-08-06 at 17 24 40

File: services/database-analyzer.js [hunk 5]
```diff
@@ -37,28 +48,49 @@ function DatabaseAnalyzer(databaseConnection, config) {
...
                 defaultValue: columnInfo.defaultValue,
               };
 
+              const typeOfField = field.primaryKey ? 'primary key' : 'field';
+              log(`  Detected ${chalk.bold(typeOfField)} ${chalk.magenta(field.cameCaseName)}`);
+              log(`    with ${chalk.bold('column name')} ${chalk.cyan(field.name)}`);
+              log(`    with ${chalk.bold('type')} ${chalk.cyan(field.type)}`);
+              if (field.defaultValue !== undefined) {
+                log(`    with ${chalk.bold('defaultValue')} ${chalk.cyan(field.defaultValue)}`);
+              }
```
Comment (arnaudbesnier) [inline services/database-analyzer.js]: one line per field would be lighter and easier to read
Something like:
```
Added model "User" (table: 'users')
  Added field "id" (column: "id", type: INTEGER)`
  Added field "fullName" (column: "full_name", type: STRING, default value: "vince")`
```

And for a better experience with mongoose, we should replace "column" by "field"

Comment (arnaudbesnier) [reply services/database-analyzer.js]: Why did you should to use 3 or 4 different colors in the logs?

File: services/database-analyzer.js [hunk 6]
```diff
@@ -37,28 +48,49 @@ function DatabaseAnalyzer(databaseConnection, config) {
...
+            if (reference.targetKey) {
+              log(`    with ${chalk.bold('target key')} ${chalk.cyan(reference.targetKey)}`);
+            }
             references.push(reference);
           } else if (type) {
             // NOTICE: If the column is of integer type, named "id" and primary, Sequelize will
             //         handle it automatically without necessary declaration.
             if (!(columnName === 'id' && type === 'INTEGER' && columnInfo.primaryKey)) {
               const field = {
+                cameCaseName: _.camelCase(columnName),
```
[the remark this answers is not in the document]
Comment (VincentMolinie) [reply services/database-analyzer.js]: I don't like the `l` 👼

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,9 +1,17 @@
 # Change Log
 
 ## [Unreleased]
+### Added
+- Command Generate - Add logs to explain what is going on.
+
+### Changed
+- Command Generate - Camelize all field names.
```
[the remark this answers is not in the document]
Comment (VincentMolinie) [reply CHANGELOG.md]: it's the `nameCamelCased`. The impact it's just when they use their fields its with the js norm, meaning as camelcase

File: services/database-analyzer.js (hunk 5 above)
Comment (VincentMolinie) [reply services/database-analyzer.js]: Just to detect easily the important information

**Comment by VincentMolinie** [comment]
[Lumber - Add logs on generate](https://trello.com/c/8tTFRX3c/3599-lumber-add-logs-on-generate)

**Comment by VincentMolinie** [comment]
Split this PR in 3:
- https://github.com/ForestAdmin/lumber/pull/243
- https://github.com/ForestAdmin/lumber/pull/241
- https://github.com/ForestAdmin/lumber/pull/240

**Pull request closed without merging**